File: 2018/day01a/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	input := read()
	answer := 0
	for _, v := range input {
		answer += v
	}
	fmt.Printf("Answer: %d\n", answer)
}

func read() (input []int) {
	s := bufio.NewScanner(os.Stdin)
	for s.Scan() {
		v, err := strconv.Atoi(s.Text())
		if err != nil {
			fmt.Printf("ERROR: read input: %v\n", err)
			continue
		}
		input = append(input, v)
	}
	return input
}

File: 2018/day01a/main.rs
use std::io;

fn main() {
	let input = read();
	let mut answer = 0;
	for v in input.iter() {
		answer += v
	}
	println!("Answer: {}", answer);
}

fn read() -> Vec<i32> {
	let mut input = Vec::new();
	loop {
		let mut v = String::new();
		io::stdin().read_line(&mut v).expect("failed to read input");
		if v.len() <= 1 {
			return input;
		}
		let v: i32 = v[..v.len() - 1].parse().expect("failed to parse input");
		input.push(v);
	}
}
